pkg/storage/ingest/reader.go

```diff
@@ -266,25 +288,37 @@ func (r *PartitionReader) switchToOngoingFetcher(ctx context.Context) {
 			// This method has been called before, no need to switch the fetcher.
 			return
```

Comment:
@gotjosh I haven't touched this part of the code, but it's it a bug? Look above. We always call `r.fetcher.Stop()` and then we check if the fetcher has changed. If it hasn't changed we exit. Shouldn't we move the `Stop()` after the `if`?

Reply:
Checking.

Reply:
It's technically not a bug because the method just no-ops [link](https://github.com/grafana/mimir/blob/main/pkg/storage/ingest/reader.go#L126) in the partition reader, but I think I agree it can be better structured as it can easily turn into a future bug if we decide to do something in that method now.

Reply:
I will fix it in a follow up PR to keep the change separated.

Reply:
Follow up PR: https://github.com/grafana/mimir/pull/9850